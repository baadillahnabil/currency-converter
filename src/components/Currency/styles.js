const styles = {
  currencyCard: {
    marginBottom: 20
  },
  detailSide: {
    width: '100%',
    padding: 10
  },
  deleteSide: {
    width: 50
  },

  cardCurrencyCode: {},
  cardAmount: {
    fontSize: 16,
    fontWeight: 'bold'
  },
  // currencyDesc: {
  //   fontStyle: 'italic',
  //   fontSize: 12
  // },
  unitRate: {
    fontSize: 12,
    fontStyle: 'italic',
    fontWeight: 'bold'
  },

  buttonDelete: {
    minWidth: 30,
    fontWeight: 'normal'
  }
}

export default styles
