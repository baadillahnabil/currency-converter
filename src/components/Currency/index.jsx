import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/styles'
import accounting from 'accounting-js'

import { Grid, Paper, Typography, Button } from '@material-ui/core'

import styles from './styles'

const Currency = ({
  classes,
  rates,
  inputAmount,
  selectedRates,
  setSelectedRates
}) => (
  <>
    {Object.keys(selectedRates).map(key => (
      <Paper key={key} elevation={1} className={classes.currencyCard}>
        <Grid container wrap="nowrap">
          {/* Conversion Detail */}
          <Grid className={classes.detailSide}>
            <Grid container justify="space-between" alignItems="baseline">
              <Typography component="p" className={classes.cardCurrencyCode}>
                {key}
              </Typography>
              <Typography component="p" className={classes.cardAmount}>
                {accounting.formatMoney(rates[key] * inputAmount, {
                  symbol: key,
                  format: '%s  %v',
                  precision: 10
                })}
              </Typography>
            </Grid>
            {/* <Typography component='p' className={classes.currencyDesc}>
                  IDR - Indonesian Rupiah // Sorry, this data ngga ada di API nya :( padahal udah dibikin
                </Typography> */}
            <Typography component="p" className={classes.unitRate}>
              1 USD ={' '}
              {accounting.formatMoney(rates[key], {
                symbol: key,
                format: '%s  %v',
                precision: 10
              })}
            </Typography>
          </Grid>

          {/* Delete Button */}
          <Button
            variant="text"
            className={classes.buttonDelete}
            onClick={() => {
              const selected = { ...selectedRates }
              delete selected[key]
              setSelectedRates(selected)
            }}
          >
            x
          </Button>
        </Grid>
      </Paper>
    ))}
  </>
)

Currency.propTypes = {
  classes: PropTypes.object.isRequired,
  rates: PropTypes.object.isRequired,
  inputAmount: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  selectedRates: PropTypes.object.isRequired,
  setSelectedRates: PropTypes.func.isRequired
}

export default withStyles(styles)(Currency)
