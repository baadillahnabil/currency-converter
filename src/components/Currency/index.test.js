/* eslint-disable no-undef */

import React from 'react'
import { shallow } from 'enzyme'

import Currency from './index'

describe('Currency Component', () => {
  it('should render correctly with all required props', () => {
    const component = shallow(
      <Currency
        rates={{}}
        inputAmount={0}
        selectedRates={{}}
        setSelectedRates={() => {}}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
