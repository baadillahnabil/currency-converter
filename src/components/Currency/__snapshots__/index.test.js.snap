// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency Component should render correctly with all required props 1`] = `ShallowWrapper {}`;
